temporalio/ext/src/util.rs

```diff
@@ -102,3 +102,28 @@ where
         *Box::from_raw(result as _)
     }
 }
+
+pub(crate) struct AsyncCallback {
+    queue: BoxValue<Value>,
+}
+
+// We trust our usage of this across threads. We would use Opaque but we can't
+// box that properly/safely.
+unsafe impl Send for AsyncCallback {}
+unsafe impl Sync for AsyncCallback {}
```

Comment:
This should explain that the inner queue `Value` is only ever supposed to be a Ruby `Queue` and that usage of that is threadsafe. The actual `Value` implementation uses a raw pointer, and hence is explicitly un-Send and Sync, but it seems like this should be fine.

Reply:
:+1: I was hoping the the method/field names were enough to clarify it is only queue. It's actually not general-purpose thread safe, heh, only in-Ruby-thread safe. They have an `Opaque` for this Ruby-only-thread thing. The problem with `Opaque` is that we can't "box" it (which basically tells Ruby not to allow it be GC'd until dropped). If they had a `BoxOpaque` our problem would be solved, heh.

Will clarify in comment here too. We may end up changing this to a `Block` or `Proc` if #162 shows that we need to do more than a queue push.